### PR TITLE
Adjust QuerySubscription::Parser for enhanced query subscription format

### PR DESCRIPTION
**Pull request by arvanasse**
Updates the query subscription parser to match the enhanced response from the reactor platform.
Also provides the 'nil fix' added to other conduit driver gems.

### CODE REVIEW

**Comment by coreypurcell** [comment]
:+1:

**Comment by arvanasse** [comment]
Merged #14

File: lib/conduit/atom/parsers/query_subscription.rb [hunk 1]
```diff
@@ -5,59 +5,67 @@ module Conduit::Driver::Atom
...
+      path = 'services/active_buckets/0/services'
+      service_object = object_path(path).detect do |service|
+        service['service'] == 'data'
+      end
+      service_object.try(:[], 'remaining_units')
     end
 
     attribute :current_plan_code do
-      object_path('current_plan_code')
+      object_path('services/active_buckets/0/plan/plan_id')
```
Comment (captainbeardo) [inline lib/conduit/atom/parsers/query_subscription.rb]: There can be more than one active bucket, so the current plan code name may be misleading.

File: lib/conduit/atom/parsers/query_subscription.rb [hunk 2]
```diff
@@ -5,59 +5,67 @@ module Conduit::Driver::Atom
...
     end
 
     attribute :nid do
-      object_path('nid')
+      object_path('carrier_data/esn_hex')
     end
 
     attribute :available_minutes do
-      object_path('available_minutes')
+      path = 'services/active_buckets/0/services'
```
Comment (captainbeardo) [inline lib/conduit/atom/parsers/query_subscription.rb]: For these to work, you'd need to loop through all the buckets and add up the values. Although there will be an enhancement to add a summary of the buckets that has this value in it. Changes can probably wait until then.

**Pull request merged**